src/GridFieldNestedForm.php

```diff
@@ -0,0 +1,435 @@
+    {
+        return ['class' => 'col-listChildrenLink grid-field__col-compact'];
+    }
+
+    public function augmentColumns($gridField, &$columns)
+    {
+        if (!in_array('ToggleNested', $columns)) {
+            array_splice($columns, 0, 0, 'ToggleNested');
+        }
+    }
+
+    public function getColumnContent($gridField, $record, $columnName)
+    {
+        $nestingLevel = $this->getNestingLevel($gridField);
+        if ($nestingLevel >= $this->getMaxNestingLevel()) {
+            return '';
+        }
+        $gridField->addExtraClass('has-nested');
+        if ($record->ID && $record->exists()) {
+            $this->gridField = $gridField;
+            $this->record = $record;
+            $relationName = $this->getRelationName();
+            if (!$record->hasMethod($relationName)) {
+                return '';
+            }
+            if ($this->canExpandCheck) {
+                if (is_callable($this->canExpandCheck)
+                    && !call_user_func($this->canExpandCheck, $record)
+                ) {
+                    return '';
+                } elseif (is_string($this->canExpandCheck)
+                    && $record->hasMethod($this->canExpandCheck)
+                    && !$this->record->{$this->canExpandCheck}($record)
+                ) {
+                    return '';
+                }
+            }
+            $toggle = 'closed';
+            $className = str_replace('\\', '-', get_class($record));
+            $state = $gridField->State->GridFieldNestedForm;
+            $stateRelation = $className.'-'.$record->ID.'-'.$this->relationName;
+            $openState = $state && (int)$state->getData($stateRelation) === 1;
+            $forceExpand = $this->expandNested && $record->$relationName()->count() > 0;
+            if (!$this->forceCloseNested
+                && ($forceExpand || $openState)
+            ) {
+                $toggle = 'open';
+            }
+
+            GridFieldExtensions::include_requirements();
+
+            return ViewableData::create()->customise([
+                'Toggle' => $toggle,
+                'Link' => $this->Link($record->ID),
+                'ToggleLink' => $this->ToggleLink($record->ID),
+                'PjaxFragment' => $stateRelation,
+                'NestedField' => ($toggle == 'open') ? $this->handleNestedItem($gridField, null, $record): ' '
+            ])->renderWith('Symbiote\GridFieldExtensions\GridFieldNestedForm');
+        }
+    }
+    
+    public function getURLHandlers($gridField)
+    {
+        return [
+            'nested/$RecordID/$NestedAction' => 'handleNestedItem',
+            'toggle/$RecordID' => 'toggleNestedItem',
+            'POST movetoparent' => 'handleMoveToParent'
+        ];
+    }
+
+    /**
+     * @param GridField $field
+     */
+    public function getHTMLFragments($field)
+    {
+        if (DataObject::has_extension($field->getModelClass(), Hierarchy::class)) {
```

Comment:
```suggestion
        if (is_a($field->getModelClass(), DataObject::class, true) && DataObject::has_extension($field->getModelClass(), Hierarchy::class)) {
```
Don't assume the modelclass is a `DataObject` subclass.
Please check the PR for other scenarios where that assumption has been made.

Reply:
Also added some checks to not assume the underlying list is Filterable, i.e. don't assume it has `filter` or `byID` functions.